### PR TITLE
feat: supply partition count for Tasklist depending on running modules

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description
Extension of https://github.com/camunda/camunda/pull/18596
Make Tasklist use `PartitionCountSupplier` to get Zeebe's parition count configuration as following:
- If the Tasklist and the Broker profiles are set, then the partition count is retrieved from the broker configuration.
- If the Tasklist and the Gateway profiles are set, then the partition count is retrieved by using the broker client that used by the Gateway to join the cluster of brokers.
- If only Tasklist is set, then the partitions count is retrieved as always by using the Zeebe client.

I also created a new maven module `webapps-common` where I moved `PartitionCountSupplier` interface.

## Related issues

relates to https://github.com/camunda/camunda/issues/18535

### CODE REVIEW

**Comment by Zelldon** [comment]
Before I look at it @romansmirnov does this conflict a bit with what you were doing? Related to the commons package? If we have access to the topology we might can use this right?

**Comment by romansmirnov** [comment]
To me, it is not contradicting. At the same time, yes, it could use the broker client to retrieve the topology and, by that, the partition count. However, to avoid blocking each other and getting Tasklist into the Single Application, I don't see this as a problem right now.

File: dist/src/main/java/io/camunda/operate/zeebe/OperatePartitionSupplierConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration(proxyBeanMethods = false)
+public class OperatePartitionSupplierConfiguration {
+  @Bean
+  public PartitionSupplier operatePartitionSupplier(
+      @Autowired(required = false) final Broker broker,
+      @Autowired(required = false) final Gateway gateway,
+      @Autowired final ZeebeClient zeebeClient) {
```
Comment (romansmirnov) [inline dist/src/main/java/io/camunda/operate/zeebe/OperatePartitionSupplierConfiguration.java:25]: @houssain-barouni, is a qualifier necessary to get the client from Operate?

Comment (houssain-barouni) [reply dist/src/main/java/io/camunda/operate/zeebe/OperatePartitionSupplierConfiguration.java:25]: It is not necessary, as the bean used by Operate is called `zeebeClient` so it takes the name of the method parameter

Comment (romansmirnov) [reply dist/src/main/java/io/camunda/operate/zeebe/OperatePartitionSupplierConfiguration.java:25]: Ah, okay, and in Tasklist, it is named differently. Got it. Thanks.

**Review by romansmirnov — APPROVED** [review]

**Pull request merged**